### PR TITLE
Configure Swagger and Refactor get order and item apis

### PR DESCRIPTION
**Pull request by zhenqixu**

### CODE REVIEW

**Comment by sandeep-nyu** [comment]
Looks like travis is not working. Are the nosetests passing?

File: service/routes.py [hunk 1]
```diff
@@ -38,16 +38,90 @@ def healthcheck():
...
+    'item_qty': fields.Integer(required=True,
+                               description='Quantity of the item'),
+    'item_price': fields.Float(required=True,
+                          description='Price of the item')
+})
+
+item_model = api.inherit(
+    'ItemModel', 
+    create_item_model,
+    {
```
Comment (sandeep-nyu) [inline service/routes.py:71]: we also need to add order_id to this like:
'order_id' : fields.Integer(readOnly=True,
                                  description='The order id that the item corresponds to'),

File: service/routes.py [hunk 2]
```diff
@@ -38,16 +38,90 @@ def healthcheck():
...
+        'id': fields.String(readOnly=True,
+                            description='The unique id assigned internally by service'),
+    }
+)
+create_model = api.model('Pet', {
```
Comment (sandeep-nyu) [inline service/routes.py:76]: Model name should be be Order

File: service/routes.py [hunk 3]
```diff
@@ -38,16 +38,90 @@ def healthcheck():
...
+    'customer_id': fields.Integer(required=True,
```
Comment (sandeep-nyu) [inline service/routes.py:77]: this should be cust_id

File: service/routes.py [hunk 4]
```diff
@@ -38,16 +38,90 @@ def healthcheck():
...
+                                  description='The customer id of the Order'),
+    'order_items': fields.List(fields.Nested(create_item_model, required=True), required=True,
```
Comment (sandeep-nyu) [inline service/routes.py:79]: I think we should have this in the order_model object below

File: service/routes.py [hunk 5]
```diff
@@ -38,16 +38,90 @@ def healthcheck():
...
+                               description='The items in the Order')
+})
+
+order_model = api.inherit(
+    'OrderModel', 
+    create_model,
+    {
+        'id': fields.String(readOnly=True,
+                            description='The unique id assigned internally by service'),
```
Comment (sandeep-nyu) [inline service/routes.py:88]: Lets make this description more explicit by saying something like : Unique order ID assigned internally by the service?

**Comment by zhenqixu** [comment]
Don't know why travis is not working. All nosetests passed.

File: service/routes.py [hunk 6]
```diff
@@ -38,16 +38,90 @@ def healthcheck():
...
+    @api.marshal_with(order_model)
+    def get(self, order_id):
+        """
+        Retrieve a single order
+        This endpoint will return an Order based on it's id
+        """
+        app.logger.info("Request for order with id: %s", order_id)
+        order = Order.find(order_id)
+        if not order:
+            api.abort(status.HTTP_404_NOT_FOUND, "Order was not found.")
```
Comment (sandeep-nyu) [inline service/routes.py:120]: As advised by Professor in class, its better to use the following abort function here, this logs and also aborts 

def abort(error_code: int, message: str):
    """Logs errors before aborting"""
    app.logger.error(message)
    api.abort(error_code, message)

File: service/routes.py [hunk 7]
```diff
@@ -157,22 +231,47 @@ def list_orders():
...
+    def get(self, order_id, item_id):
+        """
+        Read an Item in an Order
+        This endpoint will get an Item based on the order_id and item_id specified in the path
+        """
+        app.logger.info("Request to read an item in order %s with item_id: %s", order_id, item_id)
+        order = Order.find(order_id)
+
+        if not order:
+            raise NotFound("Order with id '{}' was not found.".format(order_id))
```
Comment (sandeep-nyu) [inline service/routes.py:263]: We should not use this exception here. Instead use the abort() function

File: tests/test_routes.py [hunk 8]
```diff
@@ -488,4 +489,17 @@ def test_query_by_item_id(self):
...
         # assert equal length of orders
-        self.assertEqual(len(data), len(item_id_orders))
\ No newline at end of file
+        self.assertEqual(len(data), len(item_id_orders))
+
+######################################################################
+# Utility functions
+######################################################################
+
+    def get_order(self, order):
```
Comment (sandeep-nyu) [inline tests/test_routes.py:498]: This looks like dead code to me?

**Review by sandeep-nyu — CHANGES_REQUESTED** [review]

**Review by sandeep-nyu — APPROVED** [review]

**Pull request merged**